api/src/data/modules/fields/utils.ts

```diff
@@ -592,3 +593,20 @@ export const fieldsCombinedByContentType = async ({
 
   return fields.filter(field => !(excludedNames || []).includes(field.name));
 };
+
+export const getBoardsAndPipelines = (doc: IFieldGroup) => {
+  const boardIds: string[] = [];
+  const pipelineIds: string[] = [];
+
+  for (const item of doc.boardsPipelines) {
```

Comment:
Check possible null values when looping optional fields or at least initiate it like this:
 const { boardsPipelines = [] } = doc;